qiskit/transpiler/passes/synthesis/high_level_synthesis.py

```diff
@@ -177,6 +202,17 @@ def __init__(
         self._use_qubit_indices = use_qubit_indices
         if target is not None:
             self._coupling_map = self._target.build_coupling_map()
+        self._equiv_lib = equivalence_library
+        self._basis_gates = basis_gates
+        self._min_qubits = min_qubits
+
+        self._top_level_only = top_level_only
+        if self._basis_gates is None and self._target is None:
+            self._top_level_only = True
```

Comment:
What happens if `self._top_level_only` is explicitly set to `False` and target and basis_gates are both None. Like: `HighLevelSynthesis(top_level_only=False)`. I think it's fine to ignore that, but we should document it in the docstring.

Reply:
I have decided to completely remove `top_level_only` as an argument to `HighLevelSynthesis`, at least for now, see [bcdcc28](https://github.com/Qiskit/qiskit/pull/10965/commits/bcdcc28fc6ede502555971b683c06499c0ea0afd). Initially I thought that we might need this for backward compatibility, but we don't. It is still internally computed as `self._top_level_only = self._basis_gates is None and self._target is None`, exactly as documented. I do think that eventually we might want to add even more flexibility to `HighLevelSynthesis` in the spirit of the ideas that we have discussed offline (like using HLS to change between different high-level representations of the circuit), but it's probably best not to commit to supporting `top_level_only` yet.